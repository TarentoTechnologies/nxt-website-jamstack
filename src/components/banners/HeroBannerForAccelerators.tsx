import { GatsbyImage, getImage } from "gatsby-plugin-image";

import { black50, blue, black } from "../../styles/style-guide/ColorGuide.module.css";
import {
  acceleratorDescription,
  acceleratorSubTitle,
  acceleratorTitle,
  acceleratorTitleV2,
  h3500Lato
} from "../../styles/style-guide/Typography.module.css";
import { bannerStyles } from "./HeroBannerForAccelerators.module.css";

interface HeroBannerForAcceleratorsProps {
  title: string;
  img?: any;
  subText?: string;
  description?: string;
  logosrc?: any;
}

export const HeroBannerForAccelerators = ({
  title,
  img,
  subText,
  description,
  logosrc,
}: HeroBannerForAcceleratorsProps) => {
  
  return (
    <div className={`${bannerStyles} d-flex justify-content-center`}>
      <div className="containerService paddingLeftRight15">
        <div className="row microHeading justify-content-between align-items-center">
          <div className="col-lg-6 col-md-12">
            <div className="header-info">
              <div className="">
                <img
                  src={logosrc?.localFile?.url}
                  width="217"
                  height="71"
                  alt={logosrc?.alternativeText}
                  title={logosrc?.alternativeText}
                  data-aos="fade-up"
                />
              </div>
              <h1
                className={`${acceleratorTitleV2} mt-4 mb-3 ${black}`}
                data-aos="fade-up"
                data-aos-delay="50"
              >
                {title}
              </h1>
              <div
                className={`titleSubHeading ${blue} ${h3500Lato} mb-4`}
                data-aos="fade-up"
                data-aos-delay="200"
              >
                {subText}
              </div>

              <p
                className={`${black50} ${acceleratorDescription}`}
                data-aos="fade-up"
                data-aos-delay="300"
              >
                {description}
              </p>
            </div>
          </div>
          <div className="col-lg-5 col-lg-push-1 col-md-12">
            <div className="retail-img">
              <GatsbyImage
                image={
                  getImage(img?.localFile)
                    ? getImage(img?.localFile)
                    : img?.localFile
                }
                alt={""}
                className="w-100"
                data-aos="fade-up"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
