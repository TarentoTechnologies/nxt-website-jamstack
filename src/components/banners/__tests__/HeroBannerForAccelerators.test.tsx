import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { HeroBannerForAccelerators } from "../HeroBannerForAccelerators";

const data = {
  id: "4aec6f6e-62a4-5029-9c85-331e9a91603a",
  Title: "Data. Information. Insight.",
  SubText: "Supercharge your business with the power of Data.",
  Description:
    "Modern businesses are run on information & knowledge. To improve your bottom line and expand your business, you need to have the right information available on time. Fundamentals of good data management is of paramount importance. Make reliable, data driven decisions. Take your business to the next level with our data-platform.",
  Logo: {
    alternativeText: "bolt_light",
    caption: "bolt_light",
    localFile: {
      url: "https://strapi-stage-nxt.tarento.com/uploads/bolt_light_df3a6f2e25.svg",
    },
  },
  Image: {
    alternativeText: "data-platform",
    caption: "data-platform",
    localFile: {
      childImageSharp: {
        gatsbyImageData: {
          height: 839,
          images: {
            fallback: {
              sizes: "(min-width: 786px) 786px, 100vw",
              src: "/static/8afccf7dc2c496868e9783df2c8a76b8/a4367/data_platform_2cb1ba726e.png",
              srcSet:
                "/static/8afccf7dc2c496868e9783df2c8a76b8/2a3f3/data_platform_2cb1ba726e.png 197w,\n/static/8afccf7dc2c496868e9783df2c8a76b8/fdd58/data_platform_2cb1ba726e.png 393w,\n/static/8afccf7dc2c496868e9783df2c8a76b8/a4367/data_platform_2cb1ba726e.png 786w",
            },
            layout: "constrained",
            placeholder: {
              fallback:
                "data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAVABQDASIAAhEBAxEB/8QAGAABAQEBAQAAAAAAAAAAAAAAAAIDAQT/xAAWAQEBAQAAAAAAAAAAAAAAAAABAAL/2gAMAwEAAhADEAAAAfbtN7aSCdQugv/EABoQAAMAAwEAAAAAAAAAAAAAAAECAwQSExH/2gAIAQEAAQUCedWvqdZeohuBdn5zg/abY693UUSMhFP/xAAUEQEAAAAAAAAAAAAAAAAAAAAg/9oACAEDAQE/AR//xAAXEQADAQAAAAAAAAAAAAAAAAAAAREQ/9oACAECAQE/AXRlz//EABwQAAICAwEBAAAAAAAAAAAAAAABAhESE2ExQf/aAAgBAQAGPwK1JqA0sn0pycus1USm4+GVUbrdjg/pim2f/8QAGRABAQEBAQEAAAAAAAAAAAAAAREAMSFB/9oACAEBAAE/IfYWHg4FZHj+YUZuLm6oXP1n4D0wQ9EjlcP7Pmu0AlNA4trv/9oADAMBAAIAAwAAABCDCD7/xAAWEQEBAQAAAAAAAAAAAAAAAAARASD/2gAIAQMBAT8QaGP/xAAZEQADAQEBAAAAAAAAAAAAAAAAAREhMWH/2gAIAQIBAT8Q4YRd6U54Ntus/8QAHRABAQACAwEBAQAAAAAAAAAAAREAITFBcVGBwf/aAAgBAQABPxBwkhWQke/uG9FsrQcVzfkT2ed4QThKAUvuBLCfcYDf3NI9sK6wOwQ2nAny4qIxcpG/zAlRUl35n//Z",
            },
            width: 786,
          },
        },
      },
    },
  },
};

describe("Test if the HeroBannerForAccelerators component is working as expected", () => {
  it("<HeroBannerForAccelerators /> matches snapshot", () => {
    const component = render(
      <HeroBannerForAccelerators
        title={data.Title}
        description={data.Description}
        img={data.Image}
        logosrc={data.Logo}
        subText={data.SubText}
      />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<HeroBannerForAccelerators /> renders without exploding", () => {
    const component = render(
      <HeroBannerForAccelerators
        title={data.Title}
        description={data.Description}
        img={data.Image}
        logosrc={data.Logo}
        subText={data.SubText}
      />
    );
    expect(
      component.getByTestId("hero-banner-for-accelerators")
    ).toBeInTheDocument();
  });
});
