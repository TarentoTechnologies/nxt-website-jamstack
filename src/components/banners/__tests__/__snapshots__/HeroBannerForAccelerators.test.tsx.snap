// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the HeroBannerForAccelerators component is working as expected <HeroBannerForAccelerators /> matches snapshot 1`] = `
<div>
  <div
    class="bannerStyles d-flex justify-content-center"
    data-testid="hero-banner-for-accelerators"
  >
    <div
      class="containerService paddingLeftRight15"
    >
      <div
        class="row microHeading justify-content-between align-items-center"
      >
        <div
          class="col-lg-6 col-md-12"
        >
          <div
            class="header-info"
          >
            <div
              class=""
            >
              <img
                alt="bolt_light"
                data-aos="fade-up"
                height="71"
                src="https://strapi-stage-nxt.tarento.com/uploads/bolt_light_df3a6f2e25.svg"
                title="bolt_light"
                width="217"
              />
            </div>
            <h1
              class="acceleratorTitleV2 mt-4 mb-3 black"
              data-aos="fade-up"
              data-aos-delay="50"
            >
              Data. Information. Insight.
            </h1>
            <div
              class="titleSubHeading blue h3500Lato mb-4"
              data-aos="fade-up"
              data-aos-delay="200"
            >
              Supercharge your business with the power of Data.
            </div>
            <p
              class="black50 acceleratorDescription"
              data-aos="fade-up"
              data-aos-delay="300"
            >
              Modern businesses are run on information & knowledge. To improve your bottom line and expand your business, you need to have the right information available on time. Fundamentals of good data management is of paramount importance. Make reliable, data driven decisions. Take your business to the next level with our data-platform.
            </p>
          </div>
        </div>
        <div
          class="col-lg-5 col-lg-push-1 col-md-12"
        >
          <div
            class="retail-img"
          >
            <div
              class="gatsby-image-wrapper w-100"
              data-gatsby-image-wrapper=""
              style="position: relative; overflow: hidden;"
            >
              <div
                aria-hidden="true"
                data-placeholder-image=""
                style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
              />
              <img
                alt=""
                data-aos="fade-up"
                data-gatsby-image-ssr=""
                data-main-image=""
                data-src="/static/8afccf7dc2c496868e9783df2c8a76b8/a4367/data_platform_2cb1ba726e.png"
                data-srcset="/static/8afccf7dc2c496868e9783df2c8a76b8/2a3f3/data_platform_2cb1ba726e.png 197w,/static/8afccf7dc2c496868e9783df2c8a76b8/fdd58/data_platform_2cb1ba726e.png 393w,/static/8afccf7dc2c496868e9783df2c8a76b8/a4367/data_platform_2cb1ba726e.png 786w"
                decoding="async"
                loading="lazy"
                sizes="(min-width: 786px) 786px, 100vw"
                style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
              />
              <noscript />
              <script
                type="module"
              >
                const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector("[data-placeholder-image]").style.opacity=0)}}
              </script>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
